### PR TITLE
Add child enum suffix option

### PR DESCRIPTION
**Pull request by MjnMixael**
This feature adds, hopefully, the final level of arbitrary do-whatever-you-want-ness to lua sexps. The existing `child_enum` parameter type takes whatever the parent parameter's text is and looks for a Lua Enum list with that name. It was originally intended to be able to have the parent be a Lua Enum list of Lua Enum lists. But I realized it can be whatever you want, currently. The parent could a `ship` type and then the game would look for a Lua Enum list with the name of the ship. That's handy, but it's a one-off. You can only have one list with that name.

With this upgrade a `child_enum` can optionally specify a list name suffix. So if the parent parameter is a ship, the child_enum can specify to look for a Lua Enum list that's `shipname + suffix`.

This also downgrades an Error to a log print. If the code can't find the specified enum list we don't have to hard stop. We can just return `<none>` and move on.

### CODE REVIEW

File: code/parse/sexp.cpp [hunk 1]
```diff
@@ -953,6 +955,21 @@ int get_dynamic_parameter_index(const SCP_string &op_name, int param)
 	return -1;
 }
 
+// Gets the custom suffix to append when using child enums. If we have a match
+// for operator name and parameter index, then return the suffix. Otherwise return
+// an empty string
+SCP_string get_child_enum_suffix(const SCP_string& op_name, int param_index)
+{
+	for (int i = 0; i < (int)Dynamic_enum_suffixes.size(); i++) {
```
Comment (JohnAFernandez) [inline code/parse/sexp.cpp:963]: Cast

**Review by JohnAFernandez — CHANGES_REQUESTED** [review]
Looks like just that one thing.  At least as far as I can tell.  Sexp code is not my specialty, but on the other hand, it's less than 50 lines.

File: code/parse/sexp.cpp (hunk 1 above)
Comment (MjnMixael) [reply code/parse/sexp.cpp:963]: Changed to size_t

**Review by JohnAFernandez — APPROVED** [review]

**Comment by wookieejedi** [comment]
Will plan to merge later today

**Pull request merged**